ldap/servers/slapd/auth.c

```diff
@@ -434,47 +468,84 @@ handle_handshake_done(PRFileDesc *prfd, void *clientData)
 
     if (config_get_SSLclientAuth() == SLAPD_SSLCLIENTAUTH_OFF) {
         (void)slapi_getSSLVersion_str(channelInfo.protocolVersion, sslversion, sizeof(sslversion));
-        slapi_log_access(LDAP_DEBUG_STATS, "conn=%" PRIu64 " %s %i-bit %s\n",
-                         conn->c_connid,
-                         sslversion, keySize, cipher ? cipher : "NULL");
+        if (log_format != LOG_FORMAT_DEFAULT) {
+            slapd_log_pblock_init(&logpb, log_format, NULL);
+            logpb.conn_id = conn->c_connid;
+            logpb.tls_version = sslversion;
+            logpb.keysize = keySize;
+            logpb.cipher = cipher ? cipher : "NULL";
+            slapd_log_access_tls(&logpb);
+        } else {
+            slapi_log_access(LDAP_DEBUG_STATS, "conn=%" PRIu64 " %s %i-bit %s\n",
+                    conn->c_connid,
+                    sslversion, keySize, cipher ? cipher : "NULL");
+        }
         goto done;
     }
     if (clientCert == NULL) {
         (void)slapi_getSSLVersion_str(channelInfo.protocolVersion, sslversion, sizeof(sslversion));
-        slapi_log_access(LDAP_DEBUG_STATS, "conn=%" PRIu64 " %s %i-bit %s\n",
-                         conn->c_connid,
-                         sslversion, keySize, cipher ? cipher : "NULL");
+        if (log_format != LOG_FORMAT_DEFAULT) {
+            slapd_log_pblock_init(&logpb, log_format, NULL);
+            logpb.conn_id = conn->c_connid;
+            logpb.tls_version = sslversion;
+            logpb.keysize = keySize;
+            logpb.cipher = cipher ? cipher : "NULL";
+            slapd_log_access_tls(&logpb);
+        } else {
+            slapi_log_access(LDAP_DEBUG_STATS, "conn=%" PRIu64 " %s %i-bit %s\n",
+                    conn->c_connid,
+                    sslversion, keySize, cipher ? cipher : "NULL");
+        }
```

Comment:
Should we add something like `logpb.msg = "no client certificate provided";` when `clientCert == NULL` but `config_get_SSLclientAuth() != OFF`?

Reply:
> Should we add something like `logpb.msg = "no client certificate provided";` when `clientCert == NULL` but `config_get_SSLclientAuth() != OFF`?

Yeah we can improve the logging message.

Reply:
> > Should we add something like `logpb.msg = "no client certificate provided";` when `clientCert == NULL` but `config_get_SSLclientAuth() != OFF`?
> 
> Yeah we can improve the logging message.

Actually this is not an error, and I don't think we need to state that this is not a client authentication as it's just a regular TLS connection.